inc/namespace.php

```diff
@@ -145,6 +148,26 @@ function fix_network_action_confirmation( string $url ) : string {
 	return $url;
 }
 
+/**
+ * Filter the WP Signups admin URLs to prevent endless redirects.
+ *
+ * The action handling logic in WP Signups relies on the value of `wp_get_referer()` and so
+ * does not work in cloud environments.
+ *
+ * @param string $url The full URL.
+ * @param string $admin_url The base admin URL.
+ * @param array $args The page query arguments.
+ * @return string
+ */
+function fix_user_signups_action_links( string $url, string $admin_url, array $args ) : string {
+	if ( isset( $args['action'] ) || isset( $args['bulk_action'] ) || isset( $args['bulk_action2'] ) ) {
+		$url = add_query_arg( [
+			'_wp_http_referer' => urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ),
```

Comment:
Since this is the same fix as https://github.com/humanmade/altis-cms/pull/316/files, could we kill two birds with one stone and use a single function that handles both URLs? e.g. if `$args` is passed, then we pull the action/bulk action out of that array, and otherwise we do the `wp_parse_url`?

Reply:
I considered it but there’s a subtle difference in that for the other one we’re only interested in the confirm action

Reply:
Also it throw a warning if we accepted additional arguments on the network_admin_url filter

Reply:
I wasn't thinking we'd accept additional arguments, more that those may or may not be there but they wouldn't be accepted. I saw the difference in the two functions, just trying to find ways to keep it DRY. Maybe the 

```php
		$url = add_query_arg( [
			'_wp_http_referer' => urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ),
		], $url );
```
part could be broken out, although at that point I'm not sure it's worth it. Anyway, just a thought. I don't see any other reason to not push this through.